libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/SkuInterceptor.kt

```diff
@@ -0,0 +1,23 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import android.content.Context
+import com.mapbox.services.android.navigation.v5.internal.accounts.MapboxNavigationAccounts
+import java.io.IOException
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class SkuInterceptor(private val context: Context) : Interceptor {
+
+    companion object {
+        private const val SKU_KEY = "sku"
+    }
+
+    @Throws(IOException::class)
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var request = chain.request()
+        val skuToken = MapboxNavigationAccounts.getInstance(context).obtainSkuToken()
```

Comment:
i'd recommend to inject `MapboxNavigationAccounts` via constructor instead of `Context`, makes things much more explicit and easier to test

Reply:
`MapboxNavigationAccounts` is a singleton. Why do we need to inject it? AFAIK there is no need to do it